src/phoenix/experimental/evals/functions/classify.py

```diff
@@ -343,6 +340,14 @@ def _snap_to_rail(raw_string: Optional[str], rails: List[str], verbose: bool = F
     return rail
 
 
+def _search_for_label(raw_string: str) -> str:
+    label_delimiter = r"\W*label\W*"
+    parts = re.split(label_delimiter, raw_string, maxsplit=1, flags=re.IGNORECASE)
+    if len(parts) == 2:
+        return parts[1]
+    return ""
```

Comment:
thought: we could make these types of "parsers" be injectable into `llm_classify` - because this is inherently tied to what prompt you use - if you don't use a prompt that doesn't prompt for structure things like this, you are going to want to change this.

e.g.

```python
labels_df = llm_classify(source_df, ...., label_parser: lambda (raw_str) -> str)
```

Reply:
Coming back to this after reading through everything - it's really a parser for the `with_explanation` template so maybe it should live with that template. We can default to this one but give the affordance for it to be overridden if needed.

Reply:
It's also probably worth threading through the `verbose` logging here so the end-user knows when their LLM is producing un-parsable code.